lib/irb.rb

```diff
@@ -560,6 +595,19 @@ def eval_input
           begin
             # Assignment expression check should be done before evaluate_line to handle code like `a /2#/ if false; a = 1`
             is_assignment = assignment_expression?(line)
+
+            # If the integration with debugger is activated, we need to handle certain input differently
+            if @context.with_debugger
+              command_class = load_command_class(line)
+              # First, let's pass debugging command's input to debugger
+              # Secondly, we need to let debugger evaluate non-command input
+              # Otherwise, the expression will be evaluated in the debugger's main session thread
+              # This is the only way to run the user's program in the expected thread
+              if !command_class || command_class.ancestors.include?(ExtendCommand::DebugCommand)
```

Comment:
Instead of checking the ancestors array, we can check if a class inherits from another directly.

**Note**: normally, you'd do `some_class < OtherClass`. From our learnings maintaining Tapioca, that sometimes fails for certain classes because they override `<`. So here the suggestion is inverted doing `OtherClass > some_class`, which returns the exact same result.

```suggestion
              if !command_class || ExtendCommand::DebugCommand > command_class
```

Reply:
This is nice, thanks 👍